### ISSUE TITLE
Shows No Video Found even if video exists

### ISSUE THREAD

**Issue by prakhardg** [jdepoix/youtube-transcript-api#51]
I was creating transcripts for around 10k videos using a cron. After creating around 260 transcripts, this library started giving error, video not found. However, I checked that the video existed and its CC also existed. Is there any rate limit imposed by youtube? If yes, can you describe it in detail and how can I overcome it?
Also shouldn't the library throw 429 error in that case? Please reply to this as it is urgent.

**Comment by jdepoix** [comment]
I have fetched transcripts for many thousands videos at once without running into any rate limits. Unless they recently implemented something like that, I don't think that this is the problem.
Few questions:
- Can you name some IDs it is failing for? 
- Is the error replicatable?
- does it always fail for exactly the same videos?

**Comment by prakhardg** [comment]
Please check this ID: `3sibOgsok1Q`
It says the video is no longer available, though it should say that the subtitles are disabled for this video (As it shows most of the times)

Also, please check this YouTube ID: `wehYkSa2oAg`
It gives me the same error though the captions are available. Can you provide a quick fix for me?

Here is the traceback of the error

`Traceback (most recent call last):
  File "/home/apps/haygot/content/services/transcript_service.py", line 19, in download_transcript
    transcript = YT.get_transcript(self.youtube_id, languages=['en'])
  File "/home/apps/.local/lib/python3.6/site-packages/youtube_transcript_api/_api.py", line 128, in get_transcript
    return cls.list_transcripts(video_id, proxies, cookies).find_transcript(languages).fetch()
  File "/home/apps/.local/lib/python3.6/site-packages/youtube_transcript_api/_api.py", line 70, in list_transcripts
    return TranscriptListFetcher(http_client).fetch(video_id)
  File "/home/apps/.local/lib/python3.6/site-packages/youtube_transcript_api/_transcripts.py", line 34, in fetch
    self._extract_captions_json(self._fetch_html(video_id), video_id)
  File "/home/apps/.local/lib/python3.6/site-packages/youtube_transcript_api/_transcripts.py", line 42, in _extract_captions_json
    raise VideoUnavailable(video_id)
youtube_transcript_api._errors.VideoUnavailable:
Could not retrieve a transcript for the video https://www.youtube.com/watch?v=wehYkSa2oAg! This is most likely caused by:

The video is no longer available

If you are sure that the described cause is not responsible for this error and that a transcript should be retrievable, please create an issue at https://github.com/jdepoix/youtube-transcript-api/issues. Please add which version of youtube_transcript_api you are using and provide the information needed to replicate the error. Also make sure that there are no open issues which already describe your problem!
ERROR 2020-02-25 15:44:25,797 transcript_service 24067 140381934806848
`

**Comment by jdepoix** [comment]
I can not replicate these errors.
When I run `YouTubeTranscriptApi.get_transcript('3sibOgsok1Q')` I'll get an `TranscriptsDisabled` exception. Which is correct as the transcripts for this video are disabled.
And when I run `YouTubeTranscriptApi.get_transcript('wehYkSa2oAg')`it returns the correct transcript, as expected.

Do you also run into those errors, if you just execute these requests on their own, or does it just happen if you do a lot of requests?

You can share some more code if you want and I can have a look at it, but it seems that this error is specific to your network. As I mentioned before I have also previously fetched more than 50k transcripts at once without running into any rate limits. So either theses rate limits are specific to your country or there's some kind of bottleneck in your network.

Maybe try fetching the transcript for the same video (pick one you know that it is working for) for like 10k times in a row. That way you can make sure it's not a problem with the video or API itself, but there's some kind of rate limit, if you'll start running into issues after a certain number of retries.

**Comment by prakhardg** [comment]
Okay sure will try and get back to you. Thank you for your time. Unfortunately, I cannot share the code as it is supposed to be confidential as per the organization's guidelines.

**Comment by jdepoix** [comment]
@prakhardg I will close this now, as the underlying issue doesn't seem to be related to this module. However, feel free to post updates in here, maybe they'll be helpful to someone else.

**Comment by kadedworkin** [comment]
@prakhardg Odds on you've worked through your implementation. 

Worth leaving here that from what I've been working on, I ran into the same problem. Appears that the YouTube infrastructure stops offering up responses to requests from the requesting IP for data after a certain number of queries. In my case, using a server in the SF area, pulls aren't successful after ~480 episodes in an hour and once blocked it's blocked for 24 hours. Suspect it's some kind of DDOS protection kicking in. 

To make sure it was IP based, I logged into a separate server and made the request for the target video and it worked without issue and same for my local dev machine's IP address. 

Yesterday I slowed the function down by sandbagging it with a 8 second delay at the end which effectively gets my script pulling one video every ~10.5 seconds including HTTP response time and processing of the data coming back. Gets me in the 350-380 episodes an hour range and well below the observed maximum before being blocked. Was able to collect all the episodes for a specified channel with more than 2K episodes without getting the server blocked from making pulls.
